src/circuit/circuit_frame.rs

```diff
@@ -2069,40 +2069,41 @@ fn apply_continuation<F: PrimeField, CS: ConstraintSystem<F>>(
     // Continuation::Call0
     /////////////////////////////////////////////////////////////////////////////
     let (body_form, closed_env, the_cont) = {
+        let mut call0_cs = cs.namespace(|| "Call0");
```

Comment:
I would just rebind `cs` in the scope. That would be consistent with general usage. There is always just one active namespace lexically bound to `cs`.